Source/ablastr/fields/PoissonSolver.H

```diff
@@ -49,6 +49,190 @@
+    domain.grow( phi.nGrowVect() ); // include guard cells
+
+    int nx = domain.length(0);
+    int ny = domain.length(1);
+    int nz = domain.length(2);
+
+    // Allocate 2x wider arrays for the convolution of rho with the Green function
+    // This also defines the box arrays for the global FFT: contains only one box;
+    amrex::Box const realspace_box = amrex::Box(
+        {domain.smallEnd(0), domain.smallEnd(1), domain.smallEnd(2)},
+        {2*nx-1+domain.smallEnd(0), 2*ny-1+domain.smallEnd(1), 2*nz-1+domain.smallEnd(2)},
+        amrex::IntVect::TheNodeVector() );
+    amrex::BoxArray realspace_ba = amrex::BoxArray( realspace_box );
+    amrex::Box const spectralspace_box = amrex::Box(
+        {0,0,0},
+        {nx, 2*ny-1, 2*nz-1},
+        amrex::IntVect::TheNodeVector() );
+    amrex::BoxArray spectralspace_ba = amrex::BoxArray( spectralspace_box );
+    // Define a distribution mapping for the global FFT, with only one box
+    amrex::DistributionMapping dm_global_fft;
+    dm_global_fft.define( realspace_ba );
+    // Allocate required arrays
+    amrex::MultiFab tmp_rho = amrex::MultiFab(realspace_ba, dm_global_fft, 1, 0);
+    tmp_rho.setVal(0);
+    amrex::MultiFab tmp_G = amrex::MultiFab(realspace_ba, dm_global_fft, 1, 0);
+    tmp_G.setVal(0);
+    // Allocate corresponding arrays in Fourier space
+    using SpectralField = amrex::FabArray< amrex::BaseFab <Complex> >;
+    SpectralField tmp_rho_fft = SpectralField( spectralspace_ba, dm_global_fft, 1, 0 );
+    SpectralField tmp_G_fft = SpectralField( spectralspace_ba, dm_global_fft, 1, 0 );
+
+    // Copy from rho to tmp_rho
+    tmp_rho.ParallelCopy( rho, 0, 0, 1, amrex::IntVect::TheZeroVector(), amrex::IntVect::TheZeroVector() );
+
+    // Compute the integrated Green function
+    {
+    WARPX_PROFILE("Initialize Green function");
+    amrex::BoxArray domain_ba = amrex::BoxArray( domain );
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+    for (amrex::MFIter mfi(domain_ba, dm_global_fft,amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+
+        amrex::Box bx = mfi.tilebox();
+
+        amrex::IntVect lo = realspace_box.smallEnd();
+        amrex::IntVect hi = realspace_box.bigEnd();
+
+        // Fill values of the Green function
+        amrex::Real dx = cell_size[0];
+        amrex::Real dy = cell_size[1];
+        amrex::Real dz = cell_size[2];
+        amrex::Array4<amrex::Real> tmp_G_arr = tmp_G.array(mfi);
+        amrex::ParallelFor( bx,
+            [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept
+            {
+                int i0 = i - lo[0];
+                int j0 = j - lo[1];
+                int k0 = k - lo[2];
+                amrex::Real const x = i0*dx;
+                amrex::Real const y = j0*dy;
+                amrex::Real const z = k0*dz;
+
+                amrex::Real const G_value = 1./(4*MathConst::pi*PhysConst::ep0) * (
```

Comment:
`ep::0` should be removed here, so that we can use this function also in the magnetostatic solver (same as with the multi-grid solver)